File: Javascript/Problem_Solving/LeetCode/Algorithms/BinarySearch/BinarySearchAlgorithm/BinarySearch.js
let BinarySearch = function(arr, target) 
{
    if(arr.length == 0)
        return 0;
    else 
    {
        let low = 0,
        high = arr.length - 1;
        while(low <= high)
        {
            let mid = Math.floor((high + low) / 2);
            if(arr[mid] == target)
                return mid;
            else if(arr[mid] < target)
                low = mid + 1;
            else
                high = mid - 1;
        }
        return -1;
    }
};
console.log(BinarySearch([-1,0,3,5,9,12] , 9));
console.log(BinarySearch([-1,0,3,5,9,12] , 2));
console.log(BinarySearch([-1,0,3,5,9,12] , 0));

File: C++/Problem_Solving/LeetCode/Algorithms/Algorithms01-704.BinarySearch/BinarySearch.cpp
#include <iostream>
#include <vector>
using namespace std;

int binarySearch(vector<int>& nums, int target)
{
    
    if(nums.size() == 0)
    {
        return 0;
    }
    else 
    {
        int Low = 0,
        High = nums.size() - 1;
        while (Low <= High)
        {
            int Mid = (Low + High) / 2;
            if(nums[Mid] == target)
            {
                return Mid;
            }
            else if (nums[Mid] < target)
            {
                Low = Mid + 1;
            }
            else {
                High = Mid - 1;
            }
            
        }
        
    }
}
int main(void)
{
    vector<int> arr{ 2, 3, 4, 10, 40 };
    int result = binarySearch(arr, 10);
    (result == -1)
        ? cout << "Element is not present in array"
        : cout << "Element is present at index " << result;
    return 0;
}